services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -523,22 +527,19 @@ async def delete_file(
                 if not can.delete:
                     raise FileAccessRightError(access_right="delete", file_id=file_id)
 
-        with suppress(FileMetaDataNotFoundError):
-            # NOTE: deleting might be slow, so better ensure we release the connection
-            async with self.engine.acquire() as conn:
-                file: FileMetaDataAtDB = await db_file_meta_data.get(
-                    conn, TypeAdapter(SimcoreS3FileID).validate_python(file_id)
-                )
-            await get_s3_client(self.app).delete_objects_recursively(
-                bucket=file.bucket_name,
-                prefix=(
-                    ensure_ends_with(file.file_id, "/")
-                    if file.is_directory
-                    else file.file_id
-                ),
-            )
-            async with self.engine.acquire() as conn:
-                await db_file_meta_data.delete(conn, [file.file_id])
+            enclosing_file = await find_enclosing_file(conn, user_id, file_id)
```

Comment:
this read strange to me.

1. you call `find_enclosing_file` with a given `file_id`
2. you get a "match" to your call  (i.e. `enclosing_file!=None`)

therefore, how can you still have the possibility of "not matching" that file? i.e. `enclosing_file.file_id != file_id` ??